### PR TITLE
Preview pane improvements

### PR DESCRIPTION
**Pull request by winston-de**
Improvements:
- [x] remember pane sizing
- [x] load file details even when no preview is available
- [x] cap file size for TryLoadAsText as attempting to load large files would cause performance issues
- [x] refine list of file details to load to improve performance and reduce visual clutter
- [x] fix issue where details are loaded twice when selected file is quickly switched.
- [x] improve text preview word count
- [x] make rich text preview control transparent

### CODE REVIEW

File: Files/ViewModels/SettingsViewModel.cs [hunk 1]
```diff
@@ -102,6 +102,18 @@ public GridLength SidebarWidth
             set => Set(value.Value);
         }
 
+        public GridLength PreviewPaneSize_Horizontal
```
Comment (tsvietOK) [inline Files/ViewModels/SettingsViewModel.cs:105]:
```suggestion
        public GridLength PreviewPaneSizeHorizontal
```

File: Files/Filesystem/ListedItem.cs [hunk 2]
```diff
@@ -248,7 +248,7 @@ public static string GetFriendlyDateFromFormat(DateTimeOffset d, string returnFo
             }
         }
 
-        private ObservableCollection<FileProperty> fileDetails = new ObservableCollection<FileProperty>();
+        private ObservableCollection<FileProperty> fileDetails;
```
Comment (tsvietOK) [inline Files/Filesystem/ListedItem.cs:251]: Looks like this change causes 
image

Comment (winston-de) [reply Files/Filesystem/ListedItem.cs:251]: Thanks, fixed.

**Comment by tsvietOK** [comment]
Should "No details available" be visible when shortcut selected?

**Comment by winston-de** [comment]
@tsvietOK It should be, as shortcuts aren't supported yet.
I just added a check, however it looks like ```ListedItem.IsShortcutItem``` is returning false, even if the item is a shortcut.

image

Do you have any idea as to why that may be?

**Comment by winston-de** [comment]
It looks like I can fix the issue by changing the shortcut and url check code from 
```cs
public bool IsShortcutItem => this is ShortcutItem;
public bool IsLinkItem => IsShortcutItem && ((ShortcutItem)this).IsUrl;
```
to 
```cs
public bool IsShortcutItem => FileExtension?.Equals(".lnk", StringComparison.OrdinalIgnoreCase) ?? false || (FileExtension?.Equals(".url", StringComparison.OrdinalIgnoreCase) ?? false);
public bool IsLinkItem => IsShortcutItem && (FileExtension?.Equals(".url", StringComparison.OrdinalIgnoreCase) ?? false);
```

**Comment by winston-de** [comment]
Whatever the issue is, it seems to be caused by the file caching feature, as turning it off fixes the issue.

**Review by tsvietOK — DISMISSED** [review]

File: Files/ViewModels/SettingsViewModel.cs [hunk 3]
```diff
@@ -102,6 +102,18 @@ public GridLength SidebarWidth
...
+        {
+            get => new GridLength(Math.Min(Math.Max(Get(300d), 50d), 600d), GridUnitType.Pixel);
+            set => Set(value.Value);
+        }
+        
+        public GridLength PreviewPaneSizeVertical
+        {
+            get => new GridLength(Math.Min(Math.Max(Get(200d), 50d), 600d), GridUnitType.Pixel);
+            set => Set(value.Value);
+        }
```
Comment (yaichenbaum) [inline Files/ViewModels/SettingsViewModel.cs:115]: Can you add comments for these settings?

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

File: Files/ViewModels/SettingsViewModel.cs [hunk 4]
```diff
@@ -101,13 +101,20 @@ public GridLength SidebarWidth
             get => new GridLength(Math.Min(Math.Max(Get(200d), 200d), 500d), GridUnitType.Pixel);
             set => Set(value.Value);
         }
-
+
```
Comment (yaichenbaum) [inline Files/ViewModels/SettingsViewModel.cs:104]:
```suggestion
```

File: Files/ViewModels/SettingsViewModel.cs [hunk 5]
```diff
@@ -101,13 +101,20 @@ public GridLength SidebarWidth
             get => new GridLength(Math.Min(Math.Max(Get(200d), 200d), 500d), GridUnitType.Pixel);
             set => Set(value.Value);
         }
-
+        
+        
+        /// <summary>
+        /// Gets or sets a value indicating the height that the preview pane should be opened to when the preview pane is in a horizontal layout.
```
Comment (yaichenbaum) [inline Files/ViewModels/SettingsViewModel.cs:106]:
```suggestion
        /// Gets or sets a value indicating the height of the preview pane in a horizontal layout.
```

File: Files/ViewModels/SettingsViewModel.cs [hunk 6]
```diff
@@ -101,13 +101,20 @@ public GridLength SidebarWidth
...
+        /// </summary>
         public GridLength PreviewPaneSizeHorizontal
         {
             get => new GridLength(Math.Min(Math.Max(Get(300d), 50d), 600d), GridUnitType.Pixel);
             set => Set(value.Value);
         }
         
+        /// <summary>
+        /// Gets or sets a value indicating the width that the preview pane should be opened to when the preview pane is in a vertical layout.
```
Comment (yaichenbaum) [inline Files/ViewModels/SettingsViewModel.cs:115]:
```suggestion
        /// Gets or sets a value indicating the width of the preview pane in a vertical layout.
```

File: Files/Constants.cs [hunk 7]
```diff
@@ -53,6 +53,23 @@ public static class PreviewPane
             /// The maximum number of pages loaded into the PDF preview.
             /// </summary>
             public const int PDFPageLimit = 50;
+            /// <summary>
+            /// The maximum file size, in bytes, that will attempted to be loaded as text if the extension is unkown.
```
Comment (yaichenbaum) [inline Files/Constants.cs:57]:
```suggestion
            /// The maximum file size, in bytes, that will attempted to be loaded as text if the extension is unknown.
```

Comment (winston-de) [reply Files/Constants.cs:57]: I need to get a spellchecker extension 😂

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**